src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const newState = [];
+
+  actions.map(action => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, { ...action.extraData });
+        break;
+      case 'removeProperties':
+        action.keysToRemove.map(key => delete stateCopy[key]);
+        break;
+      case 'clear':
+        Object.keys(stateCopy).forEach(key => delete stateCopy[key]);
```

Comment:
you can just assign the empty object to clear the state
```suggestion
        stateCopy = {};
```

Reply:
copy is constant so it won't work as tests said